### PR TITLE
Feature/xmldoc coin view

### PR DESCRIPTION
**Pull request by Aprogiena**
This is code style and XML doc for Stratis.Bitcoin.Features.Consensus.CoinViews

### CODE REVIEW

File: Stratis.Bitcoin/Features/Consensus/CoinViews/CachedCoinView.cs [hunk 1]
```diff
@@ -8,269 +8,355 @@
...
+        /// <summary>
+        /// Item of the coinview cache that holds information about the unspent outputs 
+        /// as well as the status of the item in relation to the underlaying storage.
+        /// </summary>
+        private class CacheItem
+        {
+            /// <summary>Information about transaction's outputs. Spent outputs are nulled.</summary>
+            public UnspentOutputs UnspentOutputs;
+
+            /// <summary><c>true</c> if the unspent output information is stored in the underlaying storage, <c>false</c> otherwise.</summary>
```
Comment (dangershony) [inline Stratis.Bitcoin/Features/Consensus/CoinViews/CachedCoinView.cs]: I would not necessarily say "underlaying storage" here, though practically it is storage but the cash coin view does not know the underline view is storage.

File: Stratis.Bitcoin/Features/Consensus/CoinViews/DBreezeCoinView.cs [hunk 2]
```diff
@@ -6,239 +6,302 @@
...
-			Guard.NotEmpty(folder, nameof(folder));
+    /// <summary>
+    /// Persistent implementation of coinview using DBreeze database.
+    /// </summary>
+    public class DBreezeCoinView : CoinView, IDisposable
+    {
+        /// <summary>Database key under which the block hash of the coin view's current tip is stored.</summary>
+        private static readonly byte[] blockHashKey = new byte[0];
+
+        /// TODO: Can we removed this? It is not used anywhere.
```
Comment (dangershony) [inline Stratis.Bitcoin/Features/Consensus/CoinViews/DBreezeCoinView.cs]: yes

File: Stratis.Bitcoin/Features/Consensus/CoinViews/FetchCoinsResponse.cs [hunk 3]
```diff
@@ -2,24 +2,35 @@
...
-		public FetchCoinsResponse(UnspentOutputs[] unspent, uint256 blockHash)
-		{
-			this.BlockHash = blockHash;
+        /// <summary>Unspent outputs of the requested transactions.</summary>
+        public UnspentOutputs[] UnspentOutputs { get; set; }
+
+        /// <summary>
+        /// Parameterless constructor.
+        /// </summary>
+        /// <remarks>TODO: This does not seem to be used anywhere by anything, can we remove it?</remarks>
```
Comment (dangershony) [inline Stratis.Bitcoin/Features/Consensus/CoinViews/FetchCoinsResponse.cs]: I guess we can remove it (but not urgent so can be done when another PR is pushed)

File: Stratis.Bitcoin/Features/Consensus/CoinViews/CachedCoinView.cs (hunk 1 above)
Comment (Aprogiena) [reply Stratis.Bitcoin/Features/Consensus/CoinViews/CachedCoinView.cs]: I know, it was intentional, and you don't like it, I change it :)

File: Stratis.Bitcoin/Features/Consensus/CoinViews/DBreezeCoinView.cs (hunk 2 above)
Comment (Aprogiena) [reply Stratis.Bitcoin/Features/Consensus/CoinViews/DBreezeCoinView.cs]: yes

File: Stratis.Bitcoin/Features/Consensus/CoinViews/FetchCoinsResponse.cs (hunk 3 above)
Comment (Aprogiena) [reply Stratis.Bitcoin/Features/Consensus/CoinViews/FetchCoinsResponse.cs]: i'll create issue then

File: Stratis.Bitcoin/Features/Consensus/CoinViews/DBreezeCoinView.cs (hunk 2 above)
Comment (Aprogiena) [reply Stratis.Bitcoin/Features/Consensus/CoinViews/DBreezeCoinView.cs]: going to the new issue as you merged this one

File: Stratis.Bitcoin/Features/Consensus/CoinViews/CachedCoinView.cs (hunk 1 above)
Comment (Aprogiena) [reply Stratis.Bitcoin/Features/Consensus/CoinViews/CachedCoinView.cs]: ok, won't, you merged it already

**Pull request merged**